shared/js/background/url-parameters.es6.js

```diff
@@ -0,0 +1,135 @@
+const Site = require('./classes/site.es6')
+const tdsStorage = require('./storage/tds.es6')
+
+/** @module */
+
+// Tracking parameters listed in the configuration are sometimes to be treated
+// as plain-text, but sometimes as regular expressions. The implementation guide
+// dictates that tracking parameter should be treated as plain-text unless they
+// contain one of the following characters: * + ? { } [ ]
+const regexpParameterTest = /[*+?{}[\]]/
+
+let plainTextTrackingParameters = null
+let regexpTrackingParameters = null
+
+function ensureTrackingParametersConfig () {
+    if (plainTextTrackingParameters && regexpTrackingParameters) {
+        return true
+    }
+
+    if (!tdsStorage.config ||
+        !tdsStorage.config.features ||
+        !tdsStorage.config.features.trackingParameters ||
+        !tdsStorage.config.features.trackingParameters.settings ||
+        !tdsStorage.config.features.trackingParameters.settings ||
```

Comment:
This line appears to be duplicated?

Reply:
Additionally can this be cleaned up using optional chaining (Or whatever its called in JS)?

e.g. `const trackingParams = tdsStorage.config?.features?.trackingParameters?.settings || null`

Reply:
> This line appears to be duplicated?

Oops, well spotted. Done.

> Additionally can this be cleaned up using optional chaining...

Unfortunately that syntax [isn't supported by Firefox < 74](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility), but we support Firefox 65+.